### ISSUE TITLE
Make mastodon browser plugins part of the mastodon project and propose their installation to users

### ISSUE THREAD

**Issue by markum** [mastodon/mastodon#22323]
### Pitch

There are a few browser plugins for Firefox and Chrome (chromium based browsers) available which ease the usage of Mastodon:
https://github.com/rugk/mastodon-simplified-federation/
https://github.com/raikasdev/mastodon4-redirect

The make it easier to interact with other instances a user browses sparing him*her to enter his home address once the user clicks to follow, to boost or fav something there. They therefor remove on of the main obstacles of using mastodon and improve the user experience a lot. As far as I understand the federated nature of mastodon does not allow making user interaction directly in the software. 

Therefor I would like to propose:
- Making the browser addons part of the official mastodon project (increasing their trustworthyness to users)
- prompting users to install a browser addon in a prominent, but non intrusive way, once a user is logged into his instance (the instance should detect the users browser and only propose the addon the user is using)
- If possible improving the installation process of the addon by telling the addon which home instance the user has, so that the user does not have to enter it manually.
- allowing the instance owners to turn this of (there will be allways people who do not liket this) 

### Motivation

I like mastodon and the idea behind it. But the normal usage via webinterface is quite a hassle and I fear part of the users will give up on this and for others it will be a frequent nuisance. Not everyone will make the switch to a mobile or desktop app. Many user will also not actively search for a browser addon. So I think it make a lot of sense to have an official browser addon which is advertised.

**Comment by CodeFoodPixels** [comment]
Instead of forcing users to make changes to their browsers (by making them install addons), we should be coming up with solutions that use the web platform itself.

Additionally, not all users want to or can install browser addons. Examples:
 - someone on an iOS device who exclusively uses mobile web rather than the app cannot install browser addons as safari does not allow this
 - browser addons have more privileges than a web page, folks may not want to install them from a security/privacy perspective

**Comment by markum** [comment]
Of course, a solution which would not require a browser addon would be prefereble. I assume, if this would be trivial, this would be already implemented. 
At the moment many users are making their frist experiences with mastodon. A first experience often sticks and the experience without those plugins is not ideal. Of course not every user wants to install a browser plugin in. But: Many users do not know about that possibility. So why do not give them the chance to discover this and increase security by having officcal addons. Everyone is still free decide.

**Comment by raikasdev** [comment]
Using addons is a bad choice for this, if it were implemented in the core.  A custom protocol (#14187) would be a better implementation
\- Creator of Mastodon4 Redirect
